src/precision/GeometryPrecisionReducer.cpp

```diff
@@ -53,6 +53,32 @@ GeometryPrecisionReducer::reduce(const Geometry& geom)
         reduced = PrecisionReducerTransformer::reduce(geom, targetPM, removeCollapsed);
     }
 
+    // Match the collection level of the output to the input
+    // if necessary
+    if (geom.isCollection()
+        && ! reduced->isCollection()
+        && (geom.getCoordinateDimension() == reduced->getCoordinateDimension())
+        && reduced->getGeometryTypeId() != GEOS_GEOMETRYCOLLECTION)
+    {
+        GeometryTypeId typeId = reduced->getGeometryTypeId();
+        if (reduced->isEmpty()) return geom.getFactory()->createEmpty(geom.getGeometryTypeId());
+        std::vector<std::unique_ptr<Geometry>> subgeoms;
+        subgeoms.push_back(std::move(reduced));
```

Comment:
Thoughts on putting this logic into a method of `GeometryFactory` ?

Reply:
Makes sense... with a no-op pass through for already-collections?